zpages/zpages.go

```diff
@@ -31,16 +31,32 @@
 package zpages // import "go.opencensus.io/zpages"
 
 import (
+	"fmt"
 	"net/http"
+	"sync"
+
+	"go.opencensus.io/internal"
 )
 
-// Handler is an http.Handler that serves the zpages.
+// Handler is deprecated: Use AppendHandlers.
 var Handler http.Handler
 
+var enableOnce sync.Once
+
 func init() {
-	zpagesMux := http.NewServeMux()
-	zpagesMux.HandleFunc("/rpcz", rpczHandler)
-	zpagesMux.HandleFunc("/tracez", tracezHandler)
-	zpagesMux.Handle("/public/", http.FileServer(fs))
-	Handler = zpagesMux
+	mux := http.NewServeMux()
+	AppendHandlers("", mux)
+	Handler = mux
+}
+
+// AppendHandlers adds the z-pages to the given ServeMux rooted at pathPrefix.
+func AppendHandlers(pathPrefix string, mux *http.ServeMux) *http.ServeMux {
```

Comment:
Since it's returning a `*ServeMux` and there's no '/' specified on the mux it's probably ok to drop the mux parameter and create it in the method. `/public` is a fs folder so additional assets could be added at run-time and (famous last words) hopefully no one would want sub-paths under `/tracez` and `/rpcz`.

What are peoples thoughts on two functions?

1. `func zpages.NewPrefixed(pathPrefix string) *ServeMux`
2. `func zpages.New() *ServeMux`

The second basically calls the first with "" and would behaviour similar to the original `zpages.Handler`.

Reply:
I think constructing our own ServeMux reduces the flexibility, in particular to add to the DefaultServeMux. Renamed the method to Handle and no longer return the ServeMux to avoid ambiguity. 

I don't think we need a separate non-prefixed call. The user can just pass an empty string explicitly, I don't think it's worth adding a convenience method because I would think that the prefixed form would actually be more common in practice.